### PR TITLE
feat(app): modify option --branch to allow release from any branch

### PR DESCRIPTION
**Pull request by leonardoanalista**
feat(app): modify option --branch to allow release from any branch

Use "*" to release from any branch.

ISSUES CLOSED: #43

Please refer to issue #43 for discussion.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,9 +10,10 @@ notifications:
 node_js:
   - '4'
   - '6'
+  - '7'
```
Comment (uglow) [inline .travis.yml]: Maybe skip v7 and test on 8?

File: README.md [hunk 2]
```diff
@@ -58,7 +58,8 @@ Of course you can change `corp-release` to any name you like.
 * `-d` or `--dryrun`: it runs in non-destructive mode. No alteration should be done in your workspace.
 * `--pre-commit [npm-script]`: Pre-commit hook. Pass the name of the npm script to run. It will run like this: `npm run [npm-script]`.
 * `--post-success [command]`: Post-success hook (after `git push` completes successfully). Pass a command to run as the argument. Eg: `--post-success "npm publish"`.
-* `-b [branch]` or `--branch [branch]`: Branch name allowed to run release. Default is `master`. If you want to release from another branch, you need to specify. 
+* `-b [branch]` or `--branch [branch]`: Branch name allowed to run release. Default is `master`. If you want to release from another branch, you need to specify.
+* `-B` or `--any-branch`: Allows release from any branch. Useful for Jenkins as git does a revision check-out.
```
Comment (uglow) [inline README.md]: I'm just wondering whether we need a separate flag for this case. What about using `-b any` or `-b *` to represent the same thing? This avoids changing the API and is still semantic IMHO.

File: .travis.yml (hunk 1 above)
Comment (leonardoanalista) [inline .travis.yml]: Sorry I meant `8`. As option `node` on line 14 means `latest stable`, I'll revert the file to original.

File: README.md (hunk 2 above)
Comment (leonardoanalista) [inline README.md]: Yes, it sounds like a good idea to me. What do you think @enriquecaballero ?
It keeps API the same and less options to document and and maintain.

**Comment by leonardoanalista** [comment]
Hi @uglow I have made the changes. Please have a look again. Please merge if you are happy with the PR.

File: src/index.js [hunk 3]
```diff
@@ -52,7 +52,7 @@ program.releasecount = (program.releasecount != undefined) ? program.releasecoun
 let version;
 
 // ### STEP 0 - Validate branch
-lib.validateBranch(program.branch);
+lib.validateBranch(program.branch, program.anyBranch);
```
Comment (uglow) [inline src/index.js]: The second argument is not required any more.

File: spec/validateBranch.spec.js [hunk 4]
```diff
@@ -46,4 +46,19 @@ describe('validateBranch', () => {
     expect(output[0]).to.include(`>>> Your release branch is: fooBranch`);
     expect(exitCalled).to.equal(false); // exit() is never called
   });
+
+  it('should show an info message an allow relrease from any branch when --any-branch is true', () => {
```
Comment (uglow) [inline spec/validateBranch.spec.js]: Typo in description: "...message an**d** allow rel~~r~~ease ..."

Comment (leonardoanalista) [inline spec/validateBranch.spec.js]: Thanks Brett. It's all done and merging after a green build

**Pull request merged**